testng-core/src/main/java/org/testng/internal/invokers/ConfigInvoker.java

```diff
@@ -374,15 +375,24 @@ private void invokeConfigurationMethod(
         testResult.setStatus(ITestResult.SUCCESS);
         return;
       }
-      if (configurableInstance != null) {
-        MethodInvocationHelper.invokeConfigurable(
-            targetInstance, params, configurableInstance, method.getMethod(), testResult);
+      boolean willfullyIgnored = false;
+      boolean usesConfigurableInstance = configurableInstance != null;
+      if (usesConfigurableInstance) {
+        willfullyIgnored =
+            !MethodInvocationHelper.invokeConfigurable(
+                targetInstance, params, configurableInstance, method.getMethod(), testResult);
       } else {
         MethodInvocationHelper.invokeMethodConsideringTimeout(
             tm, method, targetInstance, params, testResult);
       }
+      boolean testStatusRemainedUnchanged = testResult.isNotRunning();
+      if (usesConfigurableInstance && willfullyIgnored && testStatusRemainedUnchanged) {
+        throw new ConfigurationNotInvokedException(tm);
+      }
       testResult.setStatus(ITestResult.SUCCESS);
-    } catch (InvocationTargetException | IllegalAccessException ex) {
+    } catch (ConfigurationNotInvokedException
+        | InvocationTargetException
+        | IllegalAccessException ex) {
       throwConfigurationFailure(testResult, ex);
       testResult.setStatus(ITestResult.FAILURE);
       throw ex;
```

Comment:
@juherr - That was not part of the scope of this PR. Throwing exception is being done for use cases that involve a `Throwable` object as well. If we don't throw the exception then then the caller will not be able to handle skips properly. Atleast thats how the code path is.

Reply:
So, setting the status to FAILURE won't be enough?

```java
if (usesConfigurableInstance && willfullyIgnored && testStatusRemainedUnchanged) {
      throwConfigurationFailure(testResult, new ConfigurationNotInvokedException(tm));
      testResult.setStatus(ITestResult.FAILURE);
} else {
      testResult.setStatus(ITestResult.SUCCESS);
}
```

Reply:
No it wont be enough because the caller looks for exceptions to be thrown from this method to handle skips. I have tried removing this section and the tests fail.